engine/src/main/java/org/kigalisim/engine/support/RecalcKitBuilder.java

```diff
@@ -0,0 +1,85 @@
+/**
+ * Builder for creating RecalcKit instances.
+ *
+ * <p>This builder provides a fluent interface for constructing RecalcKit instances
+ * with default empty Optional values.</p>
+ *
+ * @license BSD-3-Clause
+ */
+
+package org.kigalisim.engine.support;
+
+import org.kigalisim.engine.number.UnitConverter;
+import org.kigalisim.engine.state.ConverterStateGetter;
+import org.kigalisim.engine.state.StreamKeeper;
+
+/**
+ * Builder for creating RecalcKit instances.
+ */
+public class RecalcKitBuilder {
+
+  private StreamKeeper streamKeeper;
```

Comment:
Please use Optional instead of setting and checking null for streamKeeper, unitConverter, stateGetter. Please set them to empty in the default constructor.